programmers-coding-test/sangjun/Implement/게임개발.js

```diff
@@ -0,0 +1,60 @@
+const rl = require("readline-sync");
+
+const N = rl.question("n*n 맵생성 :\n>> ").split(" ");
+const characterLocation = rl.question("캐릭터위치 바라보는방향입력 (ex 1 1 0(북)) :\n>> ").split(" ");
+
+const map = [];
+
+for (let i = 0; i < N[0]; i++) {
+    let arrayLandSee = rl.question("맵의 크기의 맞게 육지0 바다1 입력 :\n>> ").split(" ");
+    let numLandSee = arrayLandSee.map((el) => parseInt(el));
+    map.push(numLandSee);
+}
+
+const dx = [-1, 0, 1, 0];
+const dy = [0, 1, 0, -1];
+
+let x = +characterLocation[0];
+let y = +characterLocation[1];
+let direction = +characterLocation[2];
+let turnTime, nx, ny;
+turnTime, nx, ny = 0;
+let count = 1;
+
+map[x][y] = 1;
+
+function turnLeft() {
+    direction--;
+    if (direction === -1) {
+        direction = 3;
+    }
+}
+
+while (true) {
+    turnLeft();
+    nx = x + dx[direction];
+    ny = x + dy[direction];
+
+    if (turnTime === 4) {
+        nx = x - dx[direction];
+        ny = y - dy[direction];
+    }
+
+    if (map[nx][ny] === 0) {
+        map[nx][ny] = 1;
+        x = nx;
+        y = ny;
+        count++;
+        turnTime = 0;
+        console.log(count);
+        continue;
+    } else {
+        turnTime++;
+    }
```

Comment:
continue해줘서 else 필요없음. 너가 네번짼데 다 똑같은 코드라 매번 반복하는 것도 너무 귀찮네 이거.. 상준아 어뜩하냐;;
어쩜 코드들이 다 똑같지
```suggestion
    }
    turnTime++;
```